ambari-server/src/main/python/ambari_server/setupSecurity.py

```diff
@@ -716,14 +716,17 @@ def setup_ldap(options):
     err = 'Ambari Server is not running.'
     raise FatalException(1, err)
 
-  current_client_security = get_value_from_properties(properties,CLIENT_SECURITY,"no auth method")
-  if current_client_security != 'ldap':
-    query = "Currently '" + current_client_security + "' is configured, do you wish to use LDAP instead [y/n] (n)? "
-    if get_YN_input(query, False):
-      pass
-    else:
-      err = "Currently '" + current_client_security + "' configured. Can not setup LDAP."
-      raise FatalException(1, err)
+  enforce_ldap = options.ldap_force_setup if options.ldap_force_setup is not None else False
+  if not enforce_ldap:
+    current_client_security = get_value_from_properties(properties,CLIENT_SECURITY,"no auth method")
+    if current_client_security != 'ldap':
+      query = "Currently '" + current_client_security + "' is configured, do you wish to use LDAP instead [y/n] ({0})? "
+      ldap_setup_default = 'y' if current_client_security == "no auth method" else 'n'
```

Comment:
Please extract the string `"no auth method"` (appears twice) to a variable.

Reply:
Done